product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2015 OdooMRP team
+# Copyright 2015 AvanzOSC
+# Copyright 2015 Tecnativa
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import datetime
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def name_get(self):
+        res = super(ProductProduct, self.with_context(customerinfo=True)).name_get()
+        return res
+
+    @api.model
+    def _name_search(
+        self, name="", args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        res = super()._name_search(
+            name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
+        )
+        res_ids = list(res)
+        res_ids_len = len(res_ids)
+        if not limit or res_ids_len >= limit:
+            limit = (limit - res_ids_len) if limit else False
+        if (
+            not name
+            and limit
+            or not self._context.get("partner_id")
+            or res_ids_len >= limit
+        ):
+            return res_ids
+        limit -= res_ids_len
+        customerinfo_ids = self.env["product.customerinfo"]._search(
+            [
+                ("name", "=", self._context.get("partner_id")),
+                "|",
+                ("product_code", operator, name),
+                ("product_name", operator, name),
+            ],
+            limit=limit,
+            access_rights_uid=name_get_uid,
+        )
+        if not customerinfo_ids:
+            return res_ids
+        res_templates = self.browse(res_ids).mapped("product_tmpl_id")
+        product_tmpls = (
+            self.env["product.customerinfo"]
+            .browse(customerinfo_ids)
+            .mapped("product_tmpl_id")
+            - res_templates
+        )
+        product_ids = list(
+            self._search(
+                [("product_tmpl_id", "in", product_tmpls.ids)],
+                limit=limit,
+                access_rights_uid=name_get_uid,
+            )
+        )
+        res_ids.extend(product_ids)
+        return res_ids
```

Comment:
`If you are overriding name_search method in your module, you may make use now of new _rec_names_search class variable to expose the fields to search for without requiring the method override. More details at https://github.com/odoo/odoo/commit/3155c3e425581b71491844e7f9a3dd76a9f245a4.`

Maybe you can use `_rec_names_search = ['customer_ids.name', 'customer_ids.code']` for this, not sure if it works

Reply:
@hildickethan-S73 
The new function dont run with levels, in ths case with column product_tmpl_id

Reply:
![image](https://user-images.githubusercontent.com/47854752/209781842-bd885c70-3c52-4798-b2c8-9480fe966419.png)
Here it uses dot notation so I assume it can do that, would `['customer_ids.name', 'customer_ids.code', 'product_tmpl_id.customer_ids.name', 'product_tmpl_id.customer_ids.code']` not work?

Reply:
@FernandoRomera I tested this and is not working. Please, restore the _name_search override function.